### PR TITLE
DebugInterface: Display Vector Unit registers as floats

### PR DESCRIPTION
**Pull request by Souzooka**
This PR displays the values of the packed floats in the VU as floats in the debugger (using `%7.2f`), rather than the raw hexadecimal values. I'd like to provide a more precise display, but there is very limited space for text on the UI. That said, these values are likely more parse-able than just the hexadecimal values.

### CODE REVIEW

**Comment by hch12907** [comment]
Should it handle the PS2 floats? (those with exponent >=128)

**Comment by Souzooka** [comment]
I'm not familiar with that standard (and the FPU display doesn't seem to do any special handling of a different float standard).

**Comment by ramapcsx2** [comment]
I like the idea. It's probably a good help for debugging game logic.
However, this will hide emulation bugs on, for example, direct floating point comparisons.
A HEX display would make precision issues more clear then.

Could you do this as a toggle switch option? :)

**Comment by refractionpcsx2** [comment]
Also to note, values in the VF registers are not always floats, just something to be aware of.

**Comment by lightningterror** [comment]
So what is the status ?

**Comment by PSI-Rockin** [comment]
It makes sense to have a universal option saying "Display floats in hexadecimal" that acts for both COP1 and COP2. I'm thinking it would be best to place it in the context menu available via right-clicking and also bind a hotkey to it.

**Comment by ramapcsx2** [comment]
Toggle for special formatting or not, I think we can start by merging this PR first.
It's apparently popular enough that 2 people implemented it separately :p 
https://github.com/PCSX2/pcsx2/pull/3010

**Comment by lightningterror** [comment]
How about adding a Main Menu to the Debug Window, kinda like how the console has with a drop down list. Can be named Debug Options or something, this option can be in that list as a checkbox option.

**Comment by monster860** [comment]
This one gets a bit fucky if you have numbers bigger than 9999.

**Comment by ramapcsx2** [comment]
New menus need to be coded of course. It's not a problem of having them, but that someone needs to do the wxWidget stuff.

@monster860 Is it screwing up the display of the values, or the values itself? Is it still legible?

**Comment by monster860** [comment]
If I recall correctly, the printf %7.2f doesn't limit the field to 7 characters it makes the field a minimum of 7 characters. This means if, say, you have a value of -1000000000 (this exact value was actually used in a game I was debugging, believe it or not) it would show up as:

```
vf01    12.34   23.45  532.24 1234.43
vf02   -1000000000.00   23.45  532.24 1234.43
vf03    12.34   23.45  532.24 1234.43
```

In this case, vf02.x would be completely invisible.

**Comment by ramapcsx2** [comment]
Hm, maybe something can be done with the width parameter?
http://www.cplusplus.com/reference/cstdio/printf/
Otherwise, maybe make the GUI fields large enough?

**Comment by lightningterror** [comment]
Formatting needs to be adjusted.

File: pcsx2/DebugTools/DebugInterface.cpp [hunk 1]
```diff
@@ -477,6 +477,12 @@ wxString R5900DebugInterface::getRegisterString(int cat, int num)
 			sprintf(str,"%f",fpuRegs.fpr[num].f);
 			return wxString(str,wxConvUTF8);
 		}
+    case EECAT_VU0F:
+		{
+            char str[256];
+			sprintf(str, "%7.2f %7.2f %7.2f %7.2f", VU0.VF[num].F[0], VU0.VF[num].F[1], VU0.VF[num].F[2], VU0.VF[num].F[3]);
+            return wxString(str, wxConvUTF8);
```
Comment (lightningterror) [inline pcsx2/DebugTools/DebugInterface.cpp:484]: Tabs instead of spaces, above changes too.

**Comment by GovanifY** [comment]
I went ahead and rebased this to master, do we want to add anything to this PR (eg change the %7.2f, add a hex view and if so any preference?) or is it good to go?

**Comment by VelocityRa** [comment]
Can this get merged please?

**Comment by VelocityRa** [comment]
Actually, apparently you can't changed between hex & the new float view with this.
Not sure if this is acceptable considering these registers don't always have to contain floats.

**Pull request closed without merging**